lib/ddtrace/contrib/mysql2/instrumentation.rb

```diff
@@ -42,7 +42,7 @@ def query(sql, options = {})
 
           def datadog_pin
             @datadog_pin ||= Datadog::Pin.new(
-              Datadog.configuration[:mysql2][:service_name],
+              Datadog::Tracing.configuration[:mysql2][:service_name],
               app: Ext::TAG_COMPONENT,
               app_type: Datadog::Ext::AppTypes::DB,
             )
```

Comment:
Minor: Since we have a method to return the `datadog_configuration` below, I suggest using it here as well :)